csrc/scheduler/normalization_inner.cpp

```diff
@@ -96,9 +100,14 @@ bool InnerPersistentKernelScheduler::canScheduleRunTime(
 
   const int64_t warp_size = at::cuda::getCurrentDeviceProperties()->warpSize;
 
+  // check reduction properties, don't use shared memory persistent if 3D
+  // reduction
+  bool can_use_smem_persistent =
+      properties.total_reduction_numel == properties.inner_most_dimension_ndims;
```

Comment:
Shouldn't this use `inner_most_dimension_numel`?

Reply:
oh, yes, should be `inner_most_dimension_numel`

Reply:
This seems to indicate we should also have tests to validate the scheduler is indeed used when it should be.

Reply:
added.